crates/utils/src/settings/mod.rs

```diff
@@ -28,27 +14,18 @@ lazy_static! {
 
 impl Settings {
   /// Reads config from configuration file.
-  /// Then values from the environment (with prefix LEMMY) are added to the config.
-  /// And then default values are merged into config.
-  /// Defaults are controlled by Default trait implemntation for Settings structs.
   ///
   /// Note: The env var `LEMMY_DATABASE_URL` is parsed in
   /// `lemmy_db_queries/src/lib.rs::get_database_url_from_env()`
   fn init() -> Result<Self, LemmyError> {
     // Read the config file
-    let mut custom_config = from_str::<Settings>(&Self::read_config_file()?)?;
+    let config = from_str::<Settings>(&Self::read_config_file()?)?;
 
-    // Merge with env vars
-    custom_config.merge(envy::prefixed("LEMMY_").from_env::<Settings>()?);
-
-    // Merge with default
-    custom_config.merge(Settings::default());
-
-    if custom_config.hostname == Settings::default().hostname {
+    if config.hostname == "unset" {
       return Err(anyhow!("Hostname variable is not set!").into());
     }
 
-    Ok(custom_config)
+    Ok(config)
```

Comment:
You removed the merging of configs here, this means a smaller custom config won't work?

Reply:
You can try by removing lines from config/config.hjson, it runs fine after removing everything except hostname. The defaults are supplied via the `#[default(xxx)]` attributes.